src/Microsoft.TemplateEngine.TemplateLocalizer/Commands/Export/ExportCommand.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.TemplateEngine.TemplateLocalizer.Core;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer.Commands.Export
+{
+    internal sealed class ExportCommand : ModelBoundExecutableCommand<ExportCommandArgs>
+    {
+        private const string CommandName = "export";
+
+        private const int ConcurrencyLevel = 16;
+
+        public override Command CreateCommand()
+        {
+            var exportCommand = new Command(CommandName, LocalizableStrings.command_export_help_description);
+            exportCommand.AddArgument(new Argument("template-path")
+            {
+                Arity = ArgumentArity.OneOrMore,
+                ArgumentType = typeof(string),
+                Description = LocalizableStrings.command_export_help_templatePath_description,
+            });
+            exportCommand.AddOption(new Option("-r")
+            {
+                Name = "--recursive",
+                Description = LocalizableStrings.command_export_help_recursive_description,
+            });
+            exportCommand.AddOption(new Option("-l")
+            {
+                Name = "--language",
+                Description = LocalizableStrings.command_export_help_language_description,
+                Argument = new Argument()
+                {
+                    Arity = ArgumentArity.OneOrMore,
+                    ArgumentType = typeof(string),
+                },
+            });
+            exportCommand.AddOption(new Option("-d")
+            {
+                Name = "--DryRun",
+                Description = LocalizableStrings.command_export_help_dryrun_description,
+            });
+            exportCommand.Handler = this;
+
+            return exportCommand;
+        }
+
+        protected override async Task<int> Execute(ExportCommandArgs args, CancellationToken cancellationToken)
+        {
+            // Upgrade the cancellation token to allow also cancelling within this method.
+            using CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
is this necessary?  Can the token passed via method param be passed to the awaitables called in the method?

Reply:
If one of the tasks below fails, we want to cancel all the other tasks. We can't do it with a `CancellationToken` as it doesn't expose a `Cancel` method. Therefore, I created another `CancellationTokenSource` linked to the given cancellation token, which I can cancel if needed.

Reply:
I was considering this case: there is a problem with one of my templates and I'm trying to fix it, I would want to get back to my work as soon as I know about the issue (ie. without waiting for all the templates to be exported). But I guess for that case, you can invoke the tool only for that specific template as well.